src/processing/builder.rs

```diff
@@ -254,6 +260,19 @@ impl<'a> PresentationBuilder<'a> {
     }
 
     fn process_element(&mut self, element: MarkdownElement) -> Result<(), BuildError> {
+        if self.options.render_speaker_notes_only {
+            match element {
+                MarkdownElement::Comment { comment, source_position } => {
+                    self.process_comment(comment, source_position)?
+                }
+                MarkdownElement::SetexHeading { text } => self.push_slide_title(text),
```

Comment:
I was thinking showing the slide title would be good for the speaker notes mode to help match it against the title of the current slide in the main presentation.

This also means we can support implicit slide ends in the speaker notes mode for free.

Reply:
I think it would be good to have a separate `process_element` for each mode so this one's not polluted with the logic from both. But I agree that having the heading makes sense :ok_hand:

Reply:
I have split this. `process_comment` could benefit from the same split, but there are extra processing on the raw comment string.